sdk/search/Azure.Search.Documents/src/SearchExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.ClientModel.Primitives;
+using Azure.Search.Documents.Indexes;
+
+namespace Azure.Search.Documents;
+
+/// <summary>
+/// The Azure OpenAI extensions.
+/// </summary>
+public static class SearchExtensions
+{
+    /// <summary>
+    /// Gets the search client.
+    /// </summary>
+    /// <param name="provider"></param>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    public static SearchClient GetSearchClient(this ConnectionProvider provider, string indexName)
+    {
+        SearchClient searchClient = provider.Subclients.GetClient(() => CreateSearchClient(provider, indexName));
+        return searchClient;
+    }
+
+    private static SearchClient CreateSearchClient(this ConnectionProvider provider, string indexName)
+    {
+        ClientConnection connection = provider.GetConnection(typeof(SearchClient).FullName!);
+        if (!connection.TryGetLocatorAsUri(out Uri? uri) || uri is null)
+        {
+            throw new InvalidOperationException("Invalid URI.");
+        }
+        return new(uri, indexName, new AzureKeyCredential(connection.ApiKeyCredential!));
+    }
+
+    /// <summary>
+    /// Gets the search client.
+    /// </summary>
+    /// <param name="provider"></param>
+    /// <returns></returns>
+    public static SearchIndexClient GetSearchIndexClient(this ConnectionProvider provider)
+    {
+        SearchIndexClient searchIndexClient = provider.Subclients.GetClient(() => CreateSearchIndexClient(provider));
+        return searchIndexClient;
+    }
+
+    private static SearchIndexClient CreateSearchIndexClient(this ConnectionProvider provider)
+    {
+        ClientConnection connection = provider.GetConnection(typeof(SearchIndexClient).FullName!);
+        if (!connection.TryGetLocatorAsUri(out Uri? uri) || uri is null)
+        {
+            throw new InvalidOperationException("Invalid URI.");
+        }
+        return new(uri, new AzureKeyCredential(connection.ApiKeyCredential!));
+    }
+
+    /// <summary>
+    /// Gets the search client.
+    /// </summary>
+    /// <param name="provider"></param>
+    /// <returns></returns>
+    public static SearchIndexerClient GetSearchIndexerClient(this ConnectionProvider provider)
```

Comment:
@dargilco, @jhakulin, do we want this method on the project? Is it common enough scenario?

Reply:
I don't have that on the Python side yet. I'm looking at chat with RAG scenario now, and will learn more about the need for search client. So I don't have any recommendations at the moment.

Reply:
The search indexer is used to connect with a data source like Blob Storage, and it automatically crawls into the data source to update the latest data in the search index. I think it can be useful for the RAG scenario.  

For now, I will leave it as is. I will remove it if we think it is not needed.